lib/MusicBrainz/Server/Data/Relationship.pm

```diff
@@ -778,8 +780,18 @@ sub exists {
 
     $self->_check_types($type0, $type1);
 
-    my @props = qw(entity0 entity1 link_order link);
-    my @values = @{$values}{qw(entity0_id entity1_id link_order)};
+    my @props = qw(entity0 entity1);
+    my @values = @{$values}{qw(entity0_id entity1_id)};
+
+    # For part of series, do not count link order as different
+    # if everything else is the same in order to avoid duplicate additions
+    # (different number attributes still allowed)
+    my $ignore_link_order = grep { $_ eq $values->{link_type_gid} } values %PART_OF_SERIES;
```

Comment:
I see you pass `link_type_gid` to here in two places below, but if I search for `model('Relationship')->exists`, there are 5 calls (including one in the tests). So it seems kinda unclear whether it's required. Would it make sense to just add a constant for the part-of-series row IDs instead, so that a gid isn't needed?

Reply:
You mean just use `link_type_id` and have something like `@PART_OF_SERIES_ROW_IDS` that we look at instead? If that seems better, I can do that.

I'm confused about the "So it seems kinda unclear whether it's required" bit - if what is required? :) We need to look at this in here in some way to avoid a `MusicBrainz::Server::Edit::Exceptions::NoChanges->throw`, I understand.

Reply:
Seems we don't use the Perl `%PART_OF_SERIES` anywhere else, not sure if it should in that case be removed or what :)

Reply:
I pushed a commit changing to use row ids - tested quickly by disabling the JS check, seems to work fine.

Reply:
> I'm confused about the "So it seems kinda unclear whether it's required" bit - if what is required? :) We need to look at this in here in some way to avoid a MusicBrainz::Server::Edit::Exceptions::NoChanges->throw, I understand.

I meant whether the `link_type_gid` was required, since you weren't passing it in all cases. (For example, maybe because the places it wasn't being passed never dealt with series relationships.) But I prefer we always perform this check (hence switching to the row IDs).

Now I'm a bit confused: isn't this patch intended to cause *more* `NoChanges` exceptions to be thrown? (Since `NoChanges` == this is a duplicate.)